src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -427,5 +457,51 @@ private async Task AssertAutomaticReviewModifier(ClaimsPrincipal user, ReviewMod
                 throw new AuthorizationFailedException();
             }
         }
+
+        private async Task ApproveOrOutdateReviews(ClaimsPrincipal user, ReviewModel review, string revisionId)
+        {
+            //This method is called when a review is approved
+            var revisionModel = review.Revisions.Single(r => r.RevisionId == revisionId);
+            var revisionFile = revisionModel.Files.FirstOrDefault();
+            var codeFile = await _codeFileRepository.GetCodeFileAsync(revisionModel);
+
+            // Get reviews and mark them as approved if it matches current one
+            var pendingReviews = await _reviewsRepository.GetReviewsAsync(revisionFile.Language, revisionFile.PackageName, null, false);
+            foreach ( var r in pendingReviews)
+            {
+                // If review matches then mark it as approved
+                bool isReviewSame = await IsReviewSame(r, codeFile);
+                if (isReviewSame)
+                {
+                    var userId = user.GetGitHubLogin();
+                    r.Revisions.Last().Approvers.Add(userId);
+                }
+                else
+                {
+                    r.Revisions.Last().IsOutdated = true;
+                }
+                await _reviewsRepository.UpsertReviewAsync(r);
+            }
+        }
+
+        private async Task<ReviewModel> FindMatchingApprovedReview(ReviewModel review)
```

Comment:
How long does this take? Would ever Approve click take forever when we get to a higher revision count?

Reply:
This doesn't take long time since comparison us only against last revision of automatic review. But I would like this to run as background task instead of waiting for this one to update UI to avoid perf issue in the future.

Reply:
This matching logic is only triggered when scheduled automatic review is created as per latest change so this doesn't impact when approve button is clicked.